Scripts/models/logit.py

```diff
@@ -426,10 +441,32 @@ def calc_individual_mode_prob(self, is_car_user, zone):
                     else:
                         mode_exps[mode] *= math.exp(b["car_users"][1])
             mode_expsum += mode_exps[mode]
+            if mode == "car":
+                car_expsum += mode_exps[mode]
+            else:
+                sust_expsum += mode_exps[mode]
         probs = numpy.empty(len(modes))
         for i, mode in enumerate(modes):
             probs[i] = mode_exps[mode] / mode_expsum
-        return probs
+        # utils to money
+        logsum = numpy.log(mode_expsum)
+        sust_logsum = numpy.log(sust_expsum)
```

Comment:
These are expensive calculations, and I might consider some optimization even though readability would suffer. What is the run time right now?

In the current model, `sust_expsum` is the same for all agents (because car usership only modifies `car_logsum`), so we could get this from `purpose.sustainable_accessibility`.

Reply:
Assigning mode and destination for agents is around ~ 8 mins.

Getting sustainable accessibility from `purpose` works nicely and I think it is easy to follow. Also, `car_logsum ` is no longer used so it can be removed. However, this does not have much effect on running time. 

I think better efficiency would require refactoring whole `calc_individual_mode_prob` to use pre-calculated utility and logsum vectors for car users and others.